abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,1996 @@
+  - `i32 (uint32_t) max_size`
+  - `i32 (uint8_t **) return_value_data`
+  - `i32 (size_t *) return_value_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Retrieves up to` max_size` bytes starting at `offset` from the buffer
+`buffer_id`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `buffer_id`, or in case of buffer overflow
+   due to invalid `offset` and/or `max_size` values.
+- `NOT_FOUND` when requested `buffer_id` isn't available.
+- `INVALID_MEMORY_ACCESS` when `returned_value_data` and/or
+  `returned_value_size` point to invalid memory address.
+
+
+#### `proxy_get_buffer_status`
+
+* params:
+  - `i32 (`[`proxy_buffer_type_t`]`) buffer_id`
+  - `i32 (size_t *) return_buffer_size`
+  - `i32 (uint32_t *) return_ignored`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Retrieves size (`return_buffer_size`) of the buffer `buffer_id`.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `buffer_id`.
+- `NOT_FOUND` when requested `buffer_id` isn't available.
+- `INVALID_MEMORY_ACCESS` when `return_buffer_size` and/or
+  `return_ignored` point to invalid memory address.
+
+
+## HTTP fields
+
+Access to HTTP fields using functions in this section is restricted to
+specific callbacks:
+
+- `HTTP_REQUEST_HEADERS` can be read in [`proxy_on_log`],
+  and read and modified from [`proxy_on_request_headers`]
+  (or for as long as request processing is paused from it).
+- `HTTP_REQUEST_TRAILERS` can be read in [`proxy_on_log`],
+  and read and modified in [`proxy_on_request_trailers`]
+  (or for as long as request processing is paused from it).
+  They can be added in [`proxy_on_request_body`], but only when the
+  proxied request doesn't have trailers (`end_of_stream` is `true`).
+- `HTTP_RESPONSE_HEADERS` can be read in [`proxy_on_log`],
+  and read and modified in [`proxy_on_response_headers`]
+  (or for as long as response processing is paused from it).
+- `HTTP_RESPONSE_TRAILERS` can be read in [`proxy_on_log`],
+  and read and modified in [`proxy_on_response_trailers`]
+  (or for as long as response processing is paused from it).
+  They can be added in [`proxy_on_response_body`], but only when the
+  proxied response doesn't have trailers (`end_of_stream` is `true`).
+- `GRPC_CALL_INITIAL_METADATA` can be read in
+  [`proxy_on_grpc_receive_initial_metadata`].
+- `GRPC_CALL_TRAILING_METADATA` can be read in
+  [`proxy_on_grpc_receive_trailing_metadata`].
+- `HTTP_CALL_RESPONSE_HEADERS` can be read in
+  [`proxy_on_http_call_response`].
+- `HTTP_CALL_RESPONSE_TRAILERS` can be read in
+  [`proxy_on_http_call_response`].
+
+
+### Functions exposed by the host
+
+#### `proxy_get_header_map_size`
+
+* params:
+  - `i32 (`[`proxy_header_map_type_t`]`) map_id`
+  - `i32 (size_t *) return_pairs_size`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Retrieves size (`return_pairs_size`) of all key-value pairs from
+the header map `map_id`.
```

Comment:
could you clarify what `header map` mean here? I believe this includes not only HTT_HEADERS but also TRAILERS.